pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -1953,3 +1963,67 @@ def _check_unnecessary_dict_index_lookup(
                             node=subscript,
                             args=("1".join(value.as_string().rsplit("0", maxsplit=1)),),
                         )
+
+    def _check_unnecessary_list_index_lookup(
+        self, node: Union[nodes.For, nodes.Comprehension]
+    ) -> None:
+        if (
+            not isinstance(node.iter, nodes.Call)
+            or not isinstance(node.iter.func, nodes.Name)
+            or not node.iter.func.name == "enumerate"
+            or not isinstance(node.iter.args[0], nodes.Name)
+        ):
+            return
+
+        iterating_object_name = node.iter.args[0].name
+        value_variable = node.target.elts[1]
```

Comment:
Correct me if I am wrong, but wouldn't this crash for:
```python
for i in enumerate(list()):
    ...
```

I don't think we can assume `elts` has length `2`. In normal Python code we should, but `pylint` shouldn't crash when people or working on their code and lint an intermediate phase.

Could you test this and add a test case for this?